servers/src/grin/sync/header_sync.rs

```diff
@@ -66,6 +64,15 @@ impl HeaderSync {
 					header_head.height,
 				);
 
+				// Reset sync_head to header_head on transition to HeaderSync,
```

Comment:
Can you add a couple lines of comments as to what cases this covers? Maybe just an example? Helps to not forget yet again why this is here, a few months from now :-)